Docs/source/developers/testing.rst

```diff
@@ -47,97 +30,149 @@ You can run those locally before you push to save some time, install them once l
 
-For easier debugging, it can be convenient to run the tests on your local machine by executing the script
-`./run_test.sh <https://github.com/ECP-WarpX/WarpX/blob/development/run_test.sh>`__ from WarpX's root folder, as illustrated in the examples below:
+Each test directory must contain a file named ``CMakeLists.txt`` where all tests associated with the input files and scripts in that directory must be listed.
 
-.. code-block:: sh
+A new test can be added by adding a corresponding entry in ``CMakeLists.txt`` as illustrated in the examples below:
 
-    # Example:
-    # run all tests defined in ./Regression/WarpX-tests.ini
-    ./run_test.sh
+* Add the **regular test** ``test_1d_laser_acceleration``, with input file and analysis script in ``Examples/Tests/laser_acceleration/``:
 
-    # Example:
-    # run only the test named 'pml_x_yee'
-    ./run_test.sh pml_x_yee
+  .. code-block:: sh
 
-    # Example:
-    # run only the tests named 'pml_x_yee', 'pml_x_ckc' and 'pml_x_psatd'
-    ./run_test.sh pml_x_yee pml_x_ckc pml_x_psatd
+       add_warpx_test(
+           test_1d_laser_acceleration # name
+           1 # dims
+           2 # nprocs
+           OFF # eb
+           Examples/Tests/laser_acceleration/inputs_test_1d_laser_acceleration # inputs
+           Examples/Tests/laser_acceleration/analysis.py # analysis
+           diags/diag1000100 # output (plotfile)
+           OFF # dependency
+       )
 
-Note that the script `./run_test.sh <https://github.com/ECP-WarpX/WarpX/blob/development/run_test.sh>`__ runs the tests with the exact same compile-time options and runtime options used to run the tests remotely.
+* Add the **PICMI test** ``test_2d_laser_acceleration_picmi``, with input file and analysis script in ``Examples/Tests/laser_acceleration/``:
 
-Moreover, the script `./run_test.sh <https://github.com/ECP-WarpX/WarpX/blob/development/run_test.sh>`__ compiles all the executables that are necessary in order to run the chosen tests.
-The default number of threads allotted for compiling is set with ``numMakeJobs = 8`` in `./Regression/WarpX-tests.ini <https://github.com/ECP-WarpX/WarpX/blob/ad74bcbdd131a8797339ba38370b1195d0aecffb/Regression/WarpX-tests.ini#L20>`__.
-However, when running the tests on a local machine, it is usually possible and convenient to allot more threads for compiling, in order to speed up the builds.
-This can be accomplished by setting the environment variable ``WARPX_CI_NUM_MAKE_JOBS``, with the preferred number of threads that fits your local machine, e.g. ``export WARPX_CI_NUM_MAKE_JOBS=16`` (or less if your machine is smaller).
-On public CI, we overwrite the value to ``WARPX_CI_NUM_MAKE_JOBS=2``, in order to avoid overloading the available remote resources.
-Note that this will not change the number of threads used to run each test, but only the number of threads used to compile each executable necessary to run the tests.
+  .. code-block:: sh
```

Comment:
```suggestion
  .. code-block:: cmake
```

Reply:
Done in [8e64870](https://github.com/ECP-WarpX/WarpX/pull/5068/commits/8e64870fe4fe42fd6c88f69741fe7761d30226c6).